Mocking free functions from modules in another file

Is is possible to mock a module that is defined in another file?

Example
`main.rs`:
```rust
#![feature(proc_macro_hygiene)]
use mockall::automock;
use mockall::predicate::eq;
use cfg_if::cfg_if;

cfg_if! {
    if #[cfg(test)] {
        use mock_some_module::free_function;
    } else {
        use some_module::free_function;
    }
}

#[cfg_attr(test, automock)]
pub mod some_module;

fn my_func() {
    free_function(7);
}

#[test]
fn some_test() {
    let context = mock_some_module::free_function_context();
    context.expect().times(1).with(eq(7)).returning(|_x| ());
    my_func();
}
```
`some_module.rs`:
```rust
pub fn free_function(_arg: i32) {}
```

running the tests gives:
```
 --> src/main.rs:9:13
  |
9 |         use mock_some_module::free_function;
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `free_function` in `mock_some_module`

error[E0425]: cannot find function `free_function_context` in module `mock_some_module`
  --> src/main.rs:30:37
   |
30 |     let context = mock_some_module::free_function_context();
   |                                     ^^^^^^^^^^^^^^^^^^^^^ not found in `mock_some_module
```

However if I define the module directly in `main.rs` it works fine:
```rust
#[cfg_attr(test, automock)]
#[allow(dead_code)]
mod some_module {
    pub fn free_function(_arg: i32) {}
}
```

Great, thank you. That works as is, but extending a little further I'm stuck on the following scoping issue. I can open as a separate issue if that would be better. 

some_module.rs:
```rust
use cfg_if::cfg_if;
#[cfg(test)]
use mockall::automock;

pub struct SomeStruct {}

#[cfg_attr(test, automock)]
pub mod inner {
    use super::SomeStruct;
    pub fn free_function(_arg: SomeStruct) {}
}

cfg_if! {
    if #[cfg(test)] {
        pub use mock_inner::*;
    } else {
        pub use inner::*;
    }
}
```
main.rs:

```rust
#![feature(proc_macro_hygiene)]
#[cfg(test)]
use mockall::predicate::*;

pub mod some_module;
use some_module::*;

fn my_func() {
    let s = some_module::SomeStruct {};
    some_module::free_function(s);
}

fn main() {
}

#[test]
fn some_test() {
    let context = some_module::free_function_context();
    context.expect().times(1).with(always()).returning(|_x| ());
    my_func();
}
```

gives:
```
error[E0412]: cannot find type `SomeStruct` in this scope
  --> src/some_module.rs:10:32
   |
10 |     pub fn free_function(_arg: SomeStruct) {}
   |                                ^^^^^^^^^^ not found in this scope
```

...

It works if some_module.rs only is updated to only uses absolute paths:  
```rust
#[cfg_attr(test, automock)]
pub mod inner {
    pub fn free_function(_arg: crate::some_module::SomeStruct) {}
}
```

--- a/mockall_derive/src/automock.rs
+++ b/mockall_derive/src/automock.rs
@@ -545,6 +545,8 @@ fn mock_module(mod_: ItemMod) -> TokenStream {
     let items = if let Some((_, items)) = mod_.content {
         items
     } else {
+        compile_error(mod_.span(),
+        "automock can only mock inline modules, not modules from another file");
         Vec::new()
     };
     for item in items.iter() {
@@ -689,6 +691,15 @@ mod t {
         assert!(output.contains("pub ( crate ) fn expect_bang"));
         assert!(output.contains("pub ( in super :: x ) fn bean"));
         assert!(output.contains("pub ( in super :: x ) fn expect_bean"));
+    }
+
+    #[test]
+    #[should_panic(expected = "can only mock inline modules")]
+    fn external_module() {
+        let code = r#"mod foo;"#;
+        let ts = proc_macro2::TokenStream::from_str(code).unwrap();
+        let attrs_ts = proc_macro2::TokenStream::from_str("").unwrap();
+        do_automock(attrs_ts, ts).to_string();
     }
 
     #[test]

